modules/op-about-whenexpression.adoc

```diff
@@ -0,0 +1,155 @@
+// This module is included in the following assembly:
+//
+// *openshift_pipelines/understanding-openshift-pipelines.adoc
+
+[id="about-whenexpression_{context}"]
+= When expression
+
+When expressions guard task execution by setting criteria for the execution of tasks within a pipeline. They contain a list of components that allows a task to run only when certain criteria are met. When expressions are also supported in the final set of tasks that are specified using the `finally` field in the pipeline YAML file. When expressions in the `finally` task provide efficient guarded execution and improved reusability of tasks.
+
+The key components of a when expression are as follows:
+
+* `input`: Specifies static inputs or variables such as a parameter, task result, and execution status. You must enter a valid input. If you do not enter a valid input, its value defaults to an empty string.
+* `operator`: Specifies the relationship of an input to a set of `values`. Enter `in` or `notin` as your operator values.
+* `values`: Specifies an array of string values. Enter a non-empty array of static values or variables such as parameters, results, and a bound state of a workspace.
+
+The declared when expressions are evaluated before the task is run. The task is run when the value of all the when expressions is `True`. If the value of a when expression is `False`, the task is skipped.
+
+You can use the when expressions to check a lot scenarios. For example, whether:
```

Comment:
I would rather articulate this as some of when expression usecases are

Reply:
Change to: You can use the when expressions in various use cases.